packages/stryker-api/src/report/ScoreResult.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Represents a score result of a file or directory
+ */
+interface ScoreResult {
+
+  /**
+   * The file or directory name of this score result node
+   */
+  readonly name: string;
+  
+  /**
+   * Any child directory/file score result nodes
+   * If this score result represents a file, the length will be 0
+   */
+  readonly childResults: ReadonlyArray<ScoreResult>;
```

Comment:
>  I would also expect a property `parent`.
Why? A tree doesn't necessarily have a parent. You normally don't need it and would complicate the creating of the thing (cannot use a recursive function).

> I would suggest either having the methods: isFile() and isDirectory()

I understand where you're coming from, but I feel as a consumer of `ScoreResult`, you shouldn't care about whether it is a file or a folder. Take the root for example. That isn't strictly a directory, rather its a group of directories and files (which is most likely a directory, but doesn't have to be).

>  (we could over engineer this and split it into a base interface and two child interfaces for the file and directory. The directory interface could add the children property)
True, that is what i did in the html reporter :sweat: